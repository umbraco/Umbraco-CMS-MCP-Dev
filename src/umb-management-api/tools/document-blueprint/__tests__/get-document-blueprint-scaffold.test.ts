import GetDocumentBlueprintScaffoldTool from "../get/get-document-blueprint-scaffold.js";
import { DocumentBlueprintBuilder } from "./helpers/document-blueprint-builder.js";
import { DocumentBlueprintTestHelper } from "./helpers/document-blueprint-test-helper.js";
import { jest } from "@jest/globals";

const TEST_BLUEPRINT_NAME = "_Test Blueprint Scaffold";

describe("get-document-blueprint-scaffold", () => {
  let originalConsoleError: typeof console.error;

  beforeEach(() => {
    originalConsoleError = console.error;
    console.error = jest.fn();
  });

  afterEach(async () => {
    console.error = originalConsoleError;
    await DocumentBlueprintTestHelper.cleanup(TEST_BLUEPRINT_NAME);
  });

  it("should get scaffold for document blueprint", async () => {
    // Arrange: Create a document blueprint
    const builder = await new DocumentBlueprintBuilder(TEST_BLUEPRINT_NAME)
      .create();

    // Act: Get scaffold for the blueprint
    const result = await GetDocumentBlueprintScaffoldTool().handler(
      { id: builder.getId() },
      { signal: new AbortController().signal }
    );

    // Assert: Verify the response structure without snapshots due to ID normalization issues
    const responseText = result.content[0].text as string;
    const parsedResponse = JSON.parse(responseText);

    // Verify key properties exist and are correct
    expect(parsedResponse).toHaveProperty('documentType');
    expect(parsedResponse).toHaveProperty('id');
    expect(parsedResponse).toHaveProperty('values');
    expect(parsedResponse).toHaveProperty('variants');
    expect(parsedResponse.variants[0].name).toBe(TEST_BLUEPRINT_NAME);
    expect(parsedResponse.documentType.id).toBe("a95360e8-ff04-40b1-8f46-7aa4b5983096"); // ROOT_DOCUMENT_TYPE_ID
    expect(Array.isArray(parsedResponse.values)).toBe(true);
    expect(Array.isArray(parsedResponse.variants)).toBe(true);
  });

  it("should handle non-existent blueprint", async () => {
    // Act: Try to get scaffold for non-existent blueprint
    const result = await GetDocumentBlueprintScaffoldTool().handler(
      { id: "00000000-0000-0000-0000-000000000000" },
      { signal: new AbortController().signal }
    );

    // Assert: Should handle gracefully
    expect(result).toMatchSnapshot();
  });
});