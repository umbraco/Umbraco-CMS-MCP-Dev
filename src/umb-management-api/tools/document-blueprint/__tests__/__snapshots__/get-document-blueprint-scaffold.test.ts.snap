// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get-document-blueprint-scaffold should handle non-existent blueprint 1`] = `
{
  "content": [
    {
      "text": "Error using get-document-blueprint-scaffold:
{
  "message": "Request failed with status code 404",
  "response": {
    "type": "Error",
    "title": "The document blueprint could not be found",
    "status": 404,
    "operationStatus": "NotFound"
  }
}",
      "type": "text",
    },
  ],
}
`;
